File: s1ws/src/com/crystal/tigers/s1/ws/service/UserService.java
package com.crystal.tigers.s1.ws.service;

import java.util.List;

import com.crystal.tigers.s1.ws.model.User;

public interface UserService {

	User findById(int id);
	
	void saveUser(User user);
	
	void updateUser(User user);
	
	void delete(int id);

	List<User> findAllUsers(); 
	
	User findUserByName(String userName);

	boolean isUserNameUnique(Integer id, String userName);

	boolean exists(User user);

	List<User> searchByModel(User user);
}

File: s1-simple-ws/src/main/kotlin/com/crystal/tigers/s1/s1ws/services/UserService.kt
package com.crystal.tigers.s1.s1ws.services

import com.crystal.tigers.s1.s1ws.dbmodels.User

/**
 * Created by sonyynoss on 8/16/18.
 */
interface UserService {

    fun create(user: User): Boolean

    fun exists(user: User): Boolean = false
    fun updateUser(user: User) {}
    fun delete(id: Int) {}
    fun findById(id: Int): User
    fun findAllUsers(): List<User>
    fun searchByModel(user: User?): List<User>
}